File: dcp-solutions/2024-09-18/solution.php
```php
function maxNonAdjacent($nums) {
    if (empty($nums)) return 0;
    if (count($nums) <= 2) return max($nums);

    $dp = array_fill(0, count($nums), 0);
    $dp[0] = $nums[0];
    $dp[1] = max($nums[0], $nums[1]);

    for ($i = 2; $i < count($nums); $i++) {
        $dp[$i] = max($dp[$i-1], $dp[$i-2] + $nums[$i]);
    }

    return $dp[count($nums) - 1];
}
```

File: dcp-solutions/2024-09-18/solution.go
```go
func maxNonAdjacent(nums []int) int {
    if len(nums) == 0 {
        return 0
    } else if len(nums) <= 2 {
        return max(nums[0], nums[1])
    }

    prev, curr := nums[0], nums[1]
    for i := 2; i < len(nums); i++ {
        temp := curr
        curr = max(prev + nums[i], curr)
        prev = temp
    }

    return curr
}

func max(a, b int) int {
    if a > b {
        return a
    }
    return b
}
```